packages/seacas/libraries/ioss/src/text_mesh/Iotm_TextMesh.C

```diff
@@ -0,0 +1,626 @@
+  {
+    m_nodeMap[m_fillIndex] = id;
+    m_processorMap[m_fillIndex] = proc;
+
+    m_fillIndex++;
+  }
+
+  void fill_map_for_node(int64_t id, const TextMeshData& data)
+  {
+    std::set<int> procs = data.procs_for_node(id);
+
+    for (int proc : procs) {
+      if (proc != m_myProcessor) {
+        add_comm_map_pair(id, proc);
+      }
+    }
+  }
+
+  int m_myProcessor;
+  Ioss::Int64Vector &m_nodeMap;
+  std::vector<int> &m_processorMap;
+
+  size_t m_fillIndex = 0;
+};
+
+void TextMesh::node_communication_map(Ioss::Int64Vector &map, std::vector<int> &processors)
+{
+  NodeCommunicationMap commMap(m_myProcessor, map, processors);
+  commMap.verify_map_size(communication_node_count_proc());
+  commMap.fill_map_from_data(m_data);
+}
+
+void TextMesh::element_map(int64_t block_number, Ioss::Int64Vector &map) const
+{
+  raw_element_map(block_number, map);
+}
+
+void TextMesh::element_map(int64_t block_number, Ioss::IntVector &map) const
+{
+  raw_element_map(block_number, map);
+}
+
+template <typename INT>
+void TextMesh::raw_element_map(int64_t id, std::vector<INT> &map) const
+{
+  auto iter = m_blockPartition.find(id);
+  ThrowRequireMsg(iter != m_blockPartition.end(), "Could not find block with id: " << id << " in block partition");
+
+  map.reserve(iter->second.elemIds.size());
+
+  for (const auto &elemId : iter->second.elemIds) {
+    map.push_back(elemId);
+  }
+}
+
+void TextMesh::element_map(Ioss::Int64Vector &map) const
+{
+  raw_element_map(map);
+}
+
+void TextMesh::element_map(Ioss::IntVector &map) const
+{
+  raw_element_map(map);
+}
+
+template <typename INT>
+void TextMesh::raw_element_map(std::vector<INT> &map) const
+{
+  INT count = element_count_proc();
+  map.resize(count);
+
+  for (auto iter = m_blockPartition.begin(); iter != m_blockPartition.end(); iter++) {
+    size_t offset = iter->second.offset;
+    size_t blockCount = 0;
+    for (const auto &elemId : iter->second.elemIds) {
+      map[offset + blockCount++] = elemId;
+    }
+  }
+}
+
+void TextMesh::coordinates(std::vector<double> &coord) const
+{
+  /* create global coordinates */
+  int64_t count = node_count_proc();
+  coord.resize(count * m_dimension);
+  coordinates(&coord[0]);
+}
+
+void TextMesh::coordinates(double *coord) const
+{
+  if (m_rawCoordinates.empty()) return;
+
+  /* create global coordinates */
+  auto nodes = m_data.nodes_on_proc(m_myProcessor);
+  unsigned offset = 0;
+
+  for (auto node : nodes) {
+    const std::vector<double> &coords = m_coordinates[node];
+    for (unsigned i = 0; i < coords.size(); i++) {
+      coord[offset++] = coords[i];
+    }
+  }
+}
+
+void TextMesh::coordinates(std::vector<double> &x, std::vector<double> &y, std::vector<double> &z) const
+{
+  if (m_rawCoordinates.empty()) return;
+
+  /* create global coordinates */
+  int64_t count = node_count_proc();
+  x.reserve(count);
+  y.reserve(count);
+  z.reserve(count);
+
+  auto nodes = m_data.nodes_on_proc(m_myProcessor);
```

Comment:
Should this be a reference or const reference?

Reply:
Refer to previous comment